### PR TITLE
O3-2910 Incorrect Invoice Status Display for Partial Payments

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary

- Incorrect Invoice Status Display for Partial Payments

## Screenshots

https://github.com/openmrs/openmrs-esm-billing-app/assets/19533785/871a1884-f88c-41f6-802d-9228ec37b7e0

### CODE REVIEW

**Comment by CynthiaKamau** [comment]
This PR depends on the `billService` issue being fixed

**Review by ODORA0 — APPROVED** [review]

File: src/billing.resource.ts [hunk 1]
```diff
@@ -61,7 +61,12 @@ export const useBill = (billUuid: string) => {
     patientName: bill?.patient?.display.split('-')?.[1],
     identifier: bill?.patient?.display.split('-')?.[0],
     patientUuid: bill?.patient?.uuid,
-    status: bill?.status,
+    status:
+      bill.lineItems.length > 1
+        ? bill.lineItems.some((item) => item.paymentStatus === 'PENDING')
+          ? 'PENDING'
+          : 'PAID'
```
Comment (pirupius) [inline src/billing.resource.ts:68]: This lines seem to be repeated. We'll need to extract the status logic later on

File: src/invoice/payments/utils.ts [hunk 2]
```diff
@@ -1,35 +1,66 @@
...
+    billableService: getBillableServiceUuid(billableServices, lineItem.billableService),
+    item: lineItem?.item,
+    paymentStatus:
+      bill?.lineItems.length > 1
+        ? hasLineItem(selectedLineItems ?? [], lineItem)
+          ? totalAmountRendered >= lineItem.price * lineItem.quantity
+            ? 'PAID'
+            : 'PENDING'
+          : lineItem.paymentStatus
+        : lineItem.paymentStatus,
```
Comment (pirupius) [inline src/invoice/payments/utils.ts:51]: Can we extract some of this and reduce the nesting? it looks overly nested and complicated to understand

File: src/billing.resource.ts (hunk 1 above)
Comment (CynthiaKamau) [reply src/billing.resource.ts:68]: Technically the backend will handle this once the payments design is improved

File: src/invoice/payments/utils.ts (hunk 2 above)
Comment (CynthiaKamau) [reply src/invoice/payments/utils.ts:51]: What would you suggest?

**Review by pirupius — APPROVED** [review]

**Pull request merged**